crates/fuel-core/src/service/adapters.rs

```diff
@@ -87,3 +89,13 @@ impl P2PAdapter {
         Default::default()
     }
 }
+
+pub struct SyncAdapter {
+    block_rx: Receiver<Arc<ImportResult>>,
```

Comment:
I think this can be a mpsc::Receiver? There's only one service that will subscribe right? Then you want get lagged errors.

Reply:
I have used the one that the [Importer exposes](https://github.com/FuelLabs/fuel-core/blob/master/crates/services/importer/src/importer.rs#L109).

If it's only one service - then yeah we can switch the Importer code first.